src/macosMain/kotlin/co/touchlab/xcode/cli/main.kt

```diff
@@ -6,23 +6,36 @@ import co.touchlab.xcode.cli.command.Info
 import co.touchlab.xcode.cli.command.Install
 import co.touchlab.xcode.cli.command.Repair
 import co.touchlab.xcode.cli.command.Uninstall
+import co.touchlab.xcode.cli.util.CrashHelper
 import kotlinx.cli.ArgParser
 
 fun main(args: Array<String>) {
-    // If no arguments were given, we want to show help.
-    val adjustedArgs = if (args.isEmpty()) {
-        arrayOf("-h")
-    } else {
-        args
-    }
-    Logger.setLogWriters(platformLogWriter())
-    val parser = ArgParser("xcode-kotlin")
-    parser.subcommands(
-        Install(),
-        Uninstall(),
-        Repair(),
-        Info(),
-    )
+    val crashHelper = CrashHelper()
+
+    try {
+        // If no arguments were given, we want to show help.
+        val adjustedArgs = if (args.isEmpty()) {
+            arrayOf("-h")
+        } else {
+            args
+        }
+
+        Logger.setLogWriters(
+            platformLogWriter(),
+            crashHelper
+        )
 
-    parser.parse(adjustedArgs)
+        val parser = ArgParser("xcode-kotlin")
+        parser.subcommands(
+            Install(),
+            Uninstall(),
+            Repair(),
+            Info(),
+        )
+
+        parser.parse(adjustedArgs)
+    } catch (e: Throwable) {
+        Logger.e("Capturing global exception")
+        crashHelper.upload(e)
```

Comment:
We have to ask the user to upload the crash report.

```suggestion
        if (Console.confirm("xcode-kotlin has crashed, do you want to upload the crash report to Touchlab?: ")) {
            Console.echo("Uploading crash report.")
            try {
                crashHelper.upload(e)
                Console.echo("Upload successful")
                exit(1)
            } catch (uploadException: Throwable) {
                Console.printError("Uploading crash report failed!")
                e.addSupressed(uploadException)
                throw e
            }
        } else {
            throw e
        }
```

Reply:
Added - thanks!